tests/unit/synapseutils/unit_test_synapseutils_sync.py

```diff
@@ -1123,3 +1123,15 @@ def test_check_file_name_with_too_long_filename(mock_os, syn):
     assert str(ve.value) == "File name {} cannot be stored to Synapse. Names may contain letters, numbers, spaces, " \
                             "underscores, hyphens, periods, plus signs, apostrophes, " \
                             "and parentheses".format(long_file_name)
+
+
+def test_create_folder(syn):
+    folder_name = 'TestName'
+    folder_parent = 'syn123'
+    with patch.object(syn, "store") as patch_syn_store:
+        sync._create_folder(syn, folder_name, folder_parent)
+        patch_syn_store.assert_called_once_with({
+            'name': folder_name,
+            'concreteType': 'org.sagebionetworks.repo.model.Folder',
+            'parentId': folder_parent
+        })
```

Comment:
@thomasyu888: I would appreciate feedback on my first test using mocking. Looks simple, but I still wonder if there's a better approach.

Reply:
LGTM

Reply:
I could update the test like this to make sure that `_create_folder()` isn't calling `syn.store()` with the same input every time. Worth it? 

```suggestion
def test_create_folder(syn):
    folder_name = 'TestName'
    folder_parent = 'syn123'
    with patch.object(syn, "store") as patch_syn_store:
        sync._create_folder(syn, folder_name, folder_parent)
        patch_syn_store.assert_called_once_with({
            'name': folder_name,
            'concreteType': 'org.sagebionetworks.repo.model.Folder',
            'parentId': folder_parent
        })
        sync._create_folder(syn, folder_name * 2, folder_parent * 2)
        patch_syn_store.assert_called_with({
            'name': folder_name * 2,
            'concreteType': 'org.sagebionetworks.repo.model.Folder',
            'parentId': folder_parent * 2
        })
```

Reply:
I implemented this in the PR.